### PR TITLE
Material UI - Styling, Dialog, and Form Validation #40835

### PR DESCRIPTION
**Pull request by shashankbaranwal**
Task : 
1) Create a Trainee Component similar to other page component.
2) Inside Trainee Component create a folder named as components and create a new component AddDialog component
https://gyazo.com/3268e2b58ed78bd591db1ef3ce476c1a
3) AddDialog compoent will accept following props:
* open - boolean, will open the dialog if true.
* onClose - handler to close the dialog, passed from TraineeComponent
* onSubmit - handler to accept the dialog state for user: name, email, password
4) AddDialog will show information as show in the GIF
5) Add all the validations as shown in the GIF, similar to InputDemo component.
6) All the components used in the AddDialog must be from the Material UI.
7) TraineeComponent - Will hold the state of the Dialog, and update the state as per handler Request.
8) TraineeComponent - Will contain the button, and clicking on the button will open the dialog inside “AddDialog” component.

### CODE REVIEW

File: src/pages/Trainee/Trainee.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  handleClickOpen = () => {
+    this.setState({ open: true });
+  };
+
+  handleClose = () => {
+    const { open } = this.state;
+    this.setState({ open: false });
+    return open;
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/pages/Trainee/Trainee.jsx:21]: Removed return state.

File: src/pages/Trainee/components/AddDialog/Schema.jsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  Name: yup.string().trim().required('Name is a required field').min(3),
+  Email: yup.string()
+    .trim().email().required('Email is a required field'),
+  Password: yup.string()
+    .required('Password is required')
+    .matches(/^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})/, 'Must contain 8 characters, at least one uppercase letter, one lowercase letter and one number'),
+  'Confirm Password': yup.string().required('Confirm Password is required').oneOf([yup.ref('Password'), null], 'Must match password'),
+});
+
+export default schema;
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/pages/Trainee/components/AddDialog/Schema.jsx:13]: Removed Schema file for AddDialog folder and schema code added in addDialog file.

File: src/pages/Trainee/Trainee.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  handleSubmit = (data) => {
+    this.setState({
+      open: false,
+    }, () => data);
+  }
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/pages/Trainee/Trainee.jsx]: Removed callback function or console data.

File: src/pages/Trainee/components/AddDialog/Handler.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  );
+}
+Handler.propTypes = {
+  error: PropTypes.bool,
+  helperText: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+  onBlur: PropTypes.func.isRequired,
+  label: PropTypes.string,
+  type: PropTypes.string,
+  icons: PropTypes.instanceOf(Object),
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/pages/Trainee/components/AddDialog/Handler.jsx]: Removed Handler file. Add Handler to AddDialog file and update the code.

File: src/pages/Trainee/components/AddDialog/AddDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+          label={key}
+          onChange={this.handleChange(key)}
+          onBlur={() => this.isTouched(key)}
+          helperText={this.getError(key)}
+          error={!!this.getError(key)}
+          icons={constant[key]}
+          type={this.passwordType(key)}
+        />);
+      });
+
```
[the remark this answers is not in the document]
Comment (shashankbaranwal) [reply src/pages/Trainee/components/AddDialog/AddDialog.jsx]: Update AddDialog files as per the given in the task and return textfield with respect to its props.

**Review by namanpar — APPROVED** [review]

**Pull request merged**